includes/rest/class-server.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace Activitypub\Rest;
+
+use WP_REST_Response;
+use Activitypub\Signature;
+use Activitypub\Model\User;
+
+/**
+ * ActivityPub Server REST-Class
+ *
+ * @author Django Doucet
+ *
+ * @see https://www.w3.org/TR/activitypub/#security-verification
+ */
+class Server {
+
+	/**
+	 * Initialize the class, registering WordPress hooks
+	 */
+	public static function init() {
+		\add_filter( 'rest_request_before_callbacks', array( self::class, 'authorize_activitypub_requests' ), 10, 3 );
+		\add_action( 'rest_api_init', array( self::class, 'register_routes' ) );
+	}
+
+	/**
+	 * Register routes
+	 */
+	public static function register_routes() {
+		\register_rest_route(
+			'activitypub/1.0',
+			'/application',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( self::class, 'application_actor' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Render Application actor profile
+	 *
+	 * @return WP_REST_Response The JSON profile of the Application Actor.
+	 */
+	public static function application_actor() {
+		$json = new \stdClass();
+
+		$json->{'@context'} = \Activitypub\get_context();
+		$json->id = \get_rest_url( null, 'activitypub/1.0/application' );
+		$json->type = 'Application';
+		$json->preferredUsername = wp_parse_url( get_site_url(), PHP_URL_HOST ); // phpcs:ignore WordPress.NamingConventions
```

Comment:
@pfefferle 

I haven't run into any issues with this application actor so far, it has worked fine without webfinger.

The preferredUsername for this actor would be `example.com@example.com` however it seems that this could conflict with brid.gy fed.

https://fed.brid.gy/docs#profile

Reply:
Should only be an issue if you are also using brid.gy fed and I see no reason to use both plugins!?!

/cc @snarfed

Reply:
This might be an edge case:
https://fed.brid.gy/r/https://gregsplace.net/2023/05/15/saying-hello/

Reply:
What about using `example-com@example.com` or `example_com@example.com` instead?